### DISCUSSION TITLE
compile with make error: undefined reference to `pthread_*`

### DISCUSSION THREAD

**Discussion by lookou** [milvus-io/milvus#28612] [Q&A and General discussion]
```
milvus/internal/core/output/lib/libmilvus_common.so: undefined reference to `pthread_mutex_clocklock'
milvus/internal/core/output/lib/libmilvus_common.so: undefined reference to `pthread_cond_clockwait'
milvus/internal/core/output/lib/libmilvus_storage.so: undefined reference to `statx'
```

18.04.1-Ubuntu SMP
go version go1.19 linux/amd64
gcc (Ubuntu 10.3.0-1ubuntu1~18.04~1) 10.3.0

i try to add  -pthread in  script/setenv.sh but it dosen't work.

**Comment by yhmo** [comment]
I believe this error is related with gcc version or glibc version.

If your GLIBC version is lower than 2.30, try upgrade it.
This is my environment, I can build the master branch successfully.

> ldd --version                                
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
> gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
